src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -31,11 +38,25 @@ public void process(String containerName) {
 
             containerClient
                 .listBlobs()
-                .forEach(blob -> blobProcessor.process(blob.getName(), containerName));
+                .forEach(blob -> processIfReady(blob, containerName));
```

Comment:
adding `.filter()` seems more natural

Reply:
Filter was my first approach, but I went for this one because we're logging the fact of skipping the file, and having that in the predicate didn't seem right.